packages/components/src/Select.tsx

```diff
@@ -259,7 +259,25 @@ export const GroupedSelect = <
   return <ParameterizedBaseSelect findValue={findValue} value={value} options={options} {...props} />;
 };
 
-export const AsyncCreatableSelect = (props: any) => {
+export type GroupBase<Option> = RSelectGroupBase<Option>;
+
+export type AsyncCreatableSelectProps<
+  Option,
+  IsMulti extends boolean = false,
+  Group extends GroupBase<Option> = GroupBase<Option>
+> = AsyncCreatableProps<Option, IsMulti, Group>;
+
+export const AsyncCreatableSelect = <
+  Option,
```

Comment:
I don't think there is any need to duplicate the definition of `AsyncCreatableSelectProps` here given `props` being typed, but I'm also not very strong with generics. 

@msilva-broad can you take a look?

Reply:
I re-did this using `typeof` to get the necessary props type when defining `WorkspaceTagSelectProps`